src/notation/view/notationviewinputcontroller.cpp

```diff
@@ -501,20 +501,13 @@ void NotationViewInputController::mousePressEvent(QMouseEvent* event)
 
 bool NotationViewInputController::needSelect(const QMouseEvent* event, const PointF& clickLogicPos) const
 {
-    if (!event) {
+    if (!event || !hitElement()) {
         return false;
     }
 
-    const EngravingItem* hitElement = this->hitElement();
-    if (!hitElement) {
-        return false;
-    }
+    bool result = true;
 
-    Qt::MouseButton button = event->button();
-
-    bool result = hitElement && !hitElement->selected();
-
-    if (button == Qt::MouseButton::RightButton && result) {
+    if (event->button() == Qt::MouseButton::RightButton) {
         result &= !viewInteraction()->selection()->range()->containsPoint(clickLogicPos);
```

Comment:
better 'result =' instead of 'result &='

Reply:
fixed